network/src/noise/handshake.rs

```diff
@@ -248,42 +261,97 @@ impl NoiseUpgrader {
     /// that successfully authenticate to a public key in our `trusted_peers` set.
     /// In addition, we will expect the client to include an anti replay attack
     /// counter in the Noise handshake payload in mutual auth scenarios.
-    pub async fn upgrade_inbound<TSocket>(
+    pub async fn upgrade_inbound_with_noise<TSocket>(
         &self,
         mut socket: TSocket,
-    ) -> io::Result<NoiseStream<TSocket>>
+    ) -> io::Result<(NoiseStream<TSocket>, PeerId)>
     where
         TSocket: AsyncRead + AsyncWrite + Unpin,
     {
+        // receive [remote_peer_id | self_public_key] as prologue
+        let mut prologue = vec![0; PeerId::LENGTH + x25519::PUBLIC_KEY_SIZE];
```

Comment:
don't think this needs to be a vec, can just be an array?

Reply:
made the change for outbound as well!